File: src/MedEasy.RestObjects/PagedRestResponseLink.cs
namespace MedEasy.RestObjects
{
    /// <summary>
    /// Holds links in a <see cref="GenericPagedGetResponse{T}"/>
    /// </summary>
    public class PagedRestResponseLink
    {
        /// <summary>
        /// Builds a new <see cref="PagedRestResponseLink"/> instance
        /// </summary>
        /// <param name="first"></param>
        /// <param name="previous"></param>
        /// <param name="next"></param>
        /// <param name="last"></param>
        public PagedRestResponseLink(string first, string previous, string next, string last)
        {
            First = first != null 
                ? new Link { Href = first, Relation = "first"  } 
                : null;
            Next = next != null
                ? new Link { Href = next, Relation = "next" }
                : null;
            Last = last != null
                ? new Link { Href = last, Relation = "last" }
                : null;
            ;
            Previous = previous != null
                ? new Link { Href = previous, Relation = "previous" }
                : null;
            ;
        }

        /// <summary>
        /// <see cref="Link"/> to the first page of result
        /// </summary>
        public Link First { get;  }
        /// <summary>
        /// <see cref="Link"/> to the next page of result
        /// </summary>
        public Link Next { get; }
        /// <summary>
        /// <see cref="Link"/> to the last page of result
        /// </summary>
        public Link Last { get; }
        /// <summary>
        /// <see cref="Link"/> to the previous page of result
        /// </summary>
        public Link Previous { get;}
    }
}

File: src/clients/web/MedEasy.SPA/ClientApp/restObjects/PagedRestResponseLink.ts
import { Link } from "./Link";

export class PageRestResponseLink{

    public readonly first: Link;
    public readonly last: Link;
    public readonly previous: Link | null;
    public readonly next: Link | null;


    /**
     * Builds a new {PageRestResponseLink}
     * 
     * @param {string | Link} first link to the first page of result
     * @param {string | Link} previous link to the previous page of result
     * @param {string | Link} next link to the next page of result
     * @param {string | Link} last link to the last page of result
     */
    public constructor(first: string | Link, last: string | Link, previous?: string | Link, next?: string | Link) {
        this.first = typeof first === "string"
            ? Link.create(first, "first")
            : first;

        this.last = typeof last === "string"
            ? Link.create(last, "last")
            : last;

        if (previous) {
            this.previous = typeof previous === "string"
                ? Link.create(previous, "previous")
                : previous;
        }

        if (next) {
            this.next = typeof next === "string"
                ? Link.create(next, "next")
                : next;
        }

    }
}